3g-watchdog

```diff
@@ -125,28 +129,29 @@ def is_modem_present():
     return exitcode == 0
 
 
-def init_modem_power():
+def init_modem_power(pin):
     # Late import to simplify testing on developer machines.
     from RPi import GPIO
 
     GPIO.setwarnings(False)
     GPIO.setmode(GPIO.BCM)
-    GPIO.setup(MODEM_PIN, GPIO.OUT)
-    set_modem_power(True)
-
+    GPIO.setup(pin, GPIO.OUT)
+    set_modem_power(True, pin)
 
-def cycle_modem_power():
-    set_modem_power(False)
-    print("waiting...")
-    time.sleep(5)
-    set_modem_power(True)
 
+def get_cycle_modem_power(pin):
```

Comment:
Instead of using a closure it would be clearer to have a little class which wraps the modem power pin. It would have two methods name something like `set_power`, `cycle_power`. What's currently in `init_modem_power` would be in the class' `__init__`.